packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -316,15 +316,22 @@ class AdaptiveScaffold extends StatefulWidget {
     ValueChanged<int>? onDestinationSelected,
   }) {
     return Builder(
-      builder: (_) {
-        return BottomNavigationBar(
-          currentIndex: currentIndex ?? 0,
-          iconSize: iconSize,
-          items: destinations
-              .map((NavigationDestination e) => _toBottomNavItem(e))
-              .toList(),
-          onTap: onDestinationSelected,
-        );
+      builder: (BuildContext context) {
+        final NavigationBarThemeData currentNavBarTheme =
+            NavigationBarTheme.of(context);
+        return NavigationBarTheme(
+            data: currentNavBarTheme.copyWith(iconTheme:
+                MaterialStateProperty.resolveWith((Set<MaterialState> states) {
+              return currentNavBarTheme.iconTheme
+                      ?.resolve(states)
+                      ?.copyWith(size: iconSize) ??
+                  IconTheme.of(context).copyWith(size: iconSize);
+            })),
+            child: NavigationBar(
+              selectedIndex: currentIndex ?? 0,
+              destinations: destinations,
+              onDestinationSelected: onDestinationSelected,
+            ));
```

Comment:
```suggestion
        return NavigationBarTheme(
          data: currentNavBarTheme.copyWith(
            iconTheme: MaterialStateProperty.resolveWith((Set<MaterialState> states) {
              return currentNavBarTheme.iconTheme
                  ?.resolve(states)
                  ?.copyWith(size: iconSize) ??
                      IconTheme.of(context).copyWith(size: iconSize);
            }),
          ),
          child: NavigationBar(
            selectedIndex: currentIndex ?? 0,
            destinations: destinations,
            onDestinationSelected: onDestinationSelected,
          ),
        );
```

Reply:
This formatting was weird compared to the rest of the repo, I think I've corrected it here.